### ISSUE TITLE
Is CQEngine officially dead?

### ISSUE THREAD

**Issue by demwis** [npgall/cqengine#323]
I see that there is no activity on this project since last year. Seems author became inactive on Github since that time as well (hopefully he is fine and nothing serious happened in his life).

BTW, compatibility issues started to appear here (e.g. JVM 17). Highly likely some serious bugs and security issues may be identified in nearest future, especially in transitive dependencies. It becomes unsafe to use the stale project.

Is there anybody who is capable of taking care of this project in nearest future? Or may it be considered officially dead and abandoned?
If the latest, who can advise any good alternatives for it?

**Comment by lvitaly** [comment]
Yeah, there is no activity. AFAIR, there is an issue in the Kryo version, which is not supported by Java 17. If you are interested in JVM 17 support, I made those changes two years ago (see #324) and forgot to share them ☺️

**Comment by emrul** [comment]
Sad to see it neglected, it was my go-to for solving many performance bottlenecks and I just came back to it after a couple of years to investigate solving a new issue.

I will investigate how much effort it will be to replace Kryo with a more supported serializer (perhaps [eclipsestore](https://docs.eclipsestore.io/manual/serializer/index.html)?)

**Comment by lvitaly** [comment]
@emrul, it looks like Kryo is well supported (see [releases](https://github.com/EsotericSoftware/kryo/releases)). It goes a bit behind JDK releases but keeps the pace. Its CQEngine has had no Kryo updates for a long time.

But if we are talking about serializers, I think it is worth looking in [Apache Fury](https://github.com/apache/incubator-fury).

**Comment by emrul** [comment]
Well, I'm not ready to let go of this project - first priority is to keep it up to date with respect to Java versions and vulnerabilities. using @lvitaly's work as a starting point I updated it and have all tests passing on JDK21. Also updated SQLite (#319) and Guava. https://github.com/emrul/cqengine/tree/feature/jdk-21

I had to use the workaround described [here](https://github.com/jboss-javassist/javassist/issues/400#issuecomment-1113924629) to get past the restrictive reflection errors so that'll probably limit the use in some Java environments >=17 for now :-(

**Comment by emrul** [comment]
In my fork the Maven Shade plugin seems to hang now - even when running it in debug mode there's no warnings/errors.  I disabled the shade plugin and it seems to work... so more investigation required.

**Comment by ouaibsky** [comment]
Hi
I would like to provide some help on king of renovation stuff, but
@npgall could you tell us if we have a chance to get PR reviewed, then merged ? Maybe you can add other people as maintainer ?

I have in mind tasks like moving to GitHub action, jdk21 or more, maven upgrade, lib upgrades, antlr ...

Thx
Ouaibsky

**Comment by fulanya** [comment]
I believe this project has great potential, but it would be even better if it could be actively maintained with updates like those mentioned above. Any news?

**Comment by MSaifAsif** [comment]
I tried to reach out to the owner. They mentioned that they don't have enough free time at the moment to work on it.

**Comment by brinvex-dev1** [comment]
I consider CQEngine to be a very helpful and unique library, and I haven't been able to find any true alternative that matches its capabilities. It's disappointing to see it in a state of minimal maintenance, especially given its potential.

Since official updates seem unlikely, perhaps it's time for the community to take matters into its own hands. I see that some of you (@lvitaly, @emrul, @ouaibsky) have already worked on compatibility updates. Would it make sense to fork the project and continue development as a community-driven effort?

I'm willing to help in any way I can. If others are interested, maybe we can organize efforts to ensure CQEngine remains viable for modern Java versions.

**Comment by brinvex-dev1** [comment]
I consider CQEngine to be a very helpful and unique library, and I haven't been able to find any true alternative that matches its capabilities. It's disappointing to see it in a state of minimal maintenance, especially given its potential.

Since official updates seem unlikely, perhaps it's time for the community to take matters into its own hands. I see that some of you (@lvitaly, @emrul, @ouaibsky) have already worked on compatibility updates. Would it make sense to fork the project and continue development as a community-driven effort?

I'm willing to help in any way I can. If others are interested, maybe we can organize efforts to ensure CQEngine remains viable for modern Java versions.

**Comment by MSaifAsif** [comment]
Hey everyone! 👋

My friends and I have relied on this library for a long time and really love what's been built here. Since the project has been quiet lately, we've decided to launch a maintained fork to keep the engine running!

We just released version 1.0.0 under a new namespace. Our plan is to keep things secure (fixing CVEs), stay up to date with modern Java trends, and eventually add new features based on what the community needs.

You can find the updated version here: [cqengine-next](https://github.com/MSaifAsif/cqengine-next) released under my  namespace but follows the same directory layout and all so as make the transition for anyone coming over as seamless as possible.

```java
    <groupId>io.github.msaifasif</groupId>
    <artifactId>cqengine</artifactId>
    <version>1.0.0</version>
```

I am in the process of setting up github projects and some tags and labels so that we can start tracking updates, bugs, community requests and features in a bit of a structured way, any help is greatly appreciated and welcomed !

Massive thanks to npgall for the solid foundation. 

See you all over there!
